internal/security/credentials.go

```diff
@@ -47,30 +52,81 @@ func (s *SecretProvider) GetDatabaseCredentials(database db.DatabaseInfo) (commo
 // specified path will be returned.
 func (s *SecretProvider) GetSecrets(path string, keys ...string) (map[string]string, error) {
 
+	if cachedSecrets := s.getCachedSecrets(path, keys...); cachedSecrets != nil {
+		return cachedSecrets, nil
+	}
+
+	if !s.isSecurityEnabled() {
+		s.loggingClient.Debug(fmt.Sprintf("Writable.InsecureSecrets: %v", s.configuration.Writable.InsecureSecrets))
+		return s.getInsecureSecrets(path, keys...)
+	}
+
+	newSecrets, err := s.secretClient.GetSecrets(path, keys...)
+	if err != nil {
+		return nil, err
+	}
+
+	s.secrets[path] = newSecrets
```

Comment:
Is this caching the secrets? You could loose some for the cache if the path is the same but the list of keys isn't... probably should make this a func that updates the cache appropriately.

Reply:
not sure what you mean by lose some for the cache.
this just caches the secrets that are returned by the secretClient. The way secretClient works is that it returns nil if there is a key in the list that is not is actually stored at that path.
Does this address your concern?

Reply:
You are overwriting the cache, for the path,  with then result of the last call to GetSecrets. If I has two keys for that path in previous call, and this call has 1 key, then I loose the 2 key in the cache.

Reply:
good point.